library/lib_window_menu.ks

```diff
@@ -0,0 +1,91 @@
+// This script shows a menu on the terminal allowing user to
+// select one of the options and return it to calling script.
+@lazyglobal off.
+
+run lib_window.
+
+function open_window_menu{
+	parameter title.
+	parameter list_of_names.
+	parameter window.
+
+	local x is window[0].
+	local y is window[1].
+	
+	local current_option is 0.
+	local len is list_of_names:length().
+
+	local last_up is ag7.
+	local last_down is ag8.
+	local last_sel is ag9.
+	local process_state is list(
+		list(false,window,"update_window_menu",false),
+		current_option,last_up,last_down,last_sel,list_of_names,title
+	).
+	draw_window_menu(process_state).
+	return process_state.
+}
+
+function draw_window_menu{ //this is opt-in function. If other programs'
+	//developers don't want to react to terminal change, fine.
+	parameter process_state.
+
+	if not is_process_gui(process_state){
+		return.
+	}
+
+	local x is process_state[0][1][0].
+	local y is process_state[0][1][1].
+	local list_of_names is process_state[5].
+	local len is process_state[5]:length().
+	local title is process_state[6].
+	local current_option is process_state[1].
+
+	print title at(x+2,y+2).
+	local i is 0.
+	until i=len{
+		print "[ ] "+list_of_names[i] at(x+2,y+i+4).
+		set i to i+1.
+	}
+	print "7/8/9 - up/down/select" at(x+2,y+len+5).
+	//TODO: what if options dont fit on window? Scroll!
```

Comment:
In my view it's a bad idea to have a menu larger than the display area. Scrolling list should be a separate widget type. Your menu will fit nicely when the user needs to choose from a small number of hardcoded options. But choosing from a large and not hardcoded list requires a different return value type - it should be an index in the list.